script/whitelist/delegationFaucet/DelegationFaucet.sol

```diff
@@ -0,0 +1,189 @@
+import "@openzeppelin/contracts/utils/Create2.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+import "../ERC20PresetMinterPauser.sol";
+
+/**
+ * @title DelegationFaucet for M2
+ * @author Layr Labs, Inc.
+ * @notice Faucet contract setup with a dummy ERC20 token and strategy for M2 testnet release.
+ * Each operator address gets a staker contract assigned to them deterministically.
+ * This contract assumes minting role of the ERC20 token and that the ERC20 token has a whitelisted strategy.
+ */
+contract DelegationFaucet is IDelegationFaucet, Ownable {
+    IStrategyManager immutable strategyManager;
+    ERC20PresetMinterPauser immutable stakeToken;
+    IStrategy immutable stakeStrategy;
+    IDelegationManager delegation;
+
+    uint256 public constant DEFAULT_AMOUNT = 100e18;
+
+    constructor(
+        IStrategyManager _strategyManager,
+        IDelegationManager _delegation,
+        ERC20PresetMinterPauser _token,
+        IStrategy _strategy
+    ) {
+        strategyManager = _strategyManager;
+        delegation = _delegation;
+        stakeToken = _token;
+        stakeStrategy = _strategy;
+    }
+
+    /**
+     * Deploys a Staker contract if not already deployed for operator. Staker gets minted _depositAmount or
+     * DEFAULT_AMOUNT if _depositAmount is 0. Then Staker contract deposits into the strategy and delegates to operator.
+     * @param _operator The operator to delegate to
+     * @param _depositAmount The amount to deposit into the strategy
+     */
+    function mintDepositAndDelegate(address _operator, uint256 _depositAmount) public onlyOwner {
+        // Operator must be registered
+        require(delegation.isOperator(_operator), "DelegationFaucet: Operator not registered");
+        address staker = getStaker(_operator);
+        // Set deposit amount
+        if (_depositAmount == 0) {
+            _depositAmount = DEFAULT_AMOUNT;
+        }
+        stakeToken.mint(staker, _depositAmount);
+
+        // Deploy staker address if not already deployed, staker constructor will approve the StrategyManager to spend the stakeToken
+        if (!Address.isContract(staker)) {
+            Create2.deploy(
+                0,
+                bytes32(uint256(uint160(_operator))),
+                abi.encodePacked(type(DelegationFaucetStaker).creationCode, abi.encode(strategyManager, stakeToken))
+            );
+        }
+
+        // deposit into stakeToken strategy
+        uint256 shares = abi.decode(depositIntoStrategy(staker, stakeStrategy, stakeToken, _depositAmount), (uint256));
+        // increaseDelegatedShares if staker contract already delegated, o/w delegateTo operator
+        if (delegation.isDelegated(staker)) {
+            delegation.increaseDelegatedShares(staker, stakeStrategy, shares);
+        } else {
+            IDelegationManager.SignatureWithExpiry memory signatureWithExpiry;
+            delegateTo(_operator, signatureWithExpiry, bytes32(0));
+        }
```

Comment:
the 'if' part of this should revert, since the caller won't have permission to call `DelegationManager.increaseDelegatedShares` (this function is only called by the StrategyManager or EigenPodManager contracts).
the 'else' part looks correct.
I would amend this to be:
```
        if (!delegation.isDelegated(staker)) {
            IDelegationManager.SignatureWithExpiry memory signatureWithExpiry;
            delegateTo(_operator, signatureWithExpiry, bytes32(0));
        }
```

I'll note here as well that this implementation won't work if the `_operator` has set a `delegationApprover` in the DelegationManager -- it might make sense to add a `signatureWithExpiry` input to this function, instead of always using an empty struct here

Reply:
Ahh I see, looked over the contract and repo docs again. Calling `depositIntoStrategy()` which is done on L76 will already addShares to the staker and increase delegated shares if applicable. 
Will add the signature params as well.

Reply:
Added the fix to L80
```
        // delegateTo operator if not delegated
        if (!delegation.isDelegated(staker)) {
            delegateTo(_operator, _approverSignatureAndExpiry, _approverSalt);
        }